### ISSUE TITLE
Question: How to create a user via API

### ISSUE THREAD

**Issue by datenwort** [Communote/communote-server#42]
Hi,

sorry for bothering. How can I create an user via the user api call? What parameter do the post call expect?

BR

**Comment by rwi** [comment]
Although the documentation states that a POST to the Users end point is possible, it actually is not (there is probably something wrong with the generation of the documentation 😕). The REST API currently doesn't support the creation of users. The reason for this is that adding users to Communote is not a single operation but an interactive process of multiple steps: 

- self registration or invitation by another user
- confirming the email address by a click on a link in a mail
- providing details like username, name and password
- activation by an admin (if automatic activation isn't enabled)

Moreover, if an external user repository (like an LDAP directory) is used, self registration is typically disabled and users are added automatically on the first login.
We haven't  found a clean way to model these aspects in the REST API yet.

If you have some additional details about what you're trying to achieve, maybe we can find another solution 😃

**Comment by scrobbleme** [comment]
@rwi Maybe at least it should be able to still generate the user using the API and then trigger everything else (i.e. email confirmation). If something is not possible in the "standard" workflow, maybe just respond with an error.

@datenwort Workaround: write a custom API to create users.

**Comment by rwi** [comment]
@scrobbleme yes would be a way to tackle it. But whether this is actually useful for developers depends on their use-cases. If you for example need to create a working / usable user via the API, it won't help.

**Comment by datenwort** [comment]
Sorry for not answering this question since a while

@rwi thank you for the information. The use case was implementing communote into a cms as special component to communicate while other tasks will be done by other tools. But I will try to activated LDAP and SSO to avoid creating user manually. 

But wondering if the LDAP/SSO plugin will create the jsessionid automatically or not, but I will give this a try.
